lib/translate/dis/Serialization/DTOs/ModuleDto.h

```diff
@@ -9,16 +9,26 @@ namespace {
     struct ModuleDto : Serializable {
         std::string target;
         unsigned pointerSize;
-        std::vector<StructTypeDto *> structTypes;
+        StructTypeDto *directoryStreamType;
+        StructTypeDto *directoryEntryType;
+        StructTypeDto *jumpBufferType;
+        StructTypeDto *localeType;
+        StructTypeDto *statType;
+        StructTypeDto *threadType;
+        StructTypeDto *vaListType;
         std::vector<FunctionDto *> functions;
         std::vector<GlobalDto *> globals;
 
-        explicit ModuleDto(Module *module, DataLayout *dataLayout)
+        explicit ModuleDto(LLVMContext &context, Module *module, DataLayout *dataLayout)
                 : target(module->getTargetTriple()),
-                  pointerSize(dataLayout->getPointerSizeInBits()) {
-            for (auto &structType : module->getIdentifiedStructTypes()) {
```

Comment:
This entire PR makes the world slightly worse, in my opinion, but it is a prerequisite to escape from the local minimum in which we currently find ourselves. A subsequent PR should remove all the c++ which is obviously a massive net win despite this ugliness. The issue is that the `getIdentifiedStructTypes` method here does not appear to have an equivalent in the `LLVMSharp` bindings, so I have to get the ones I need individually by name. Eventually I hope to get rid of all of these names entirely by improving how we get parameter and return types elsewhere, but for now this is the relatively small change that will get us where we need to be.